### ISSUE TITLE
Add generality security decorator instead of @ApiBearerAuth() @ApiOAuth2Auth()

### ISSUE THREAD

**Issue by ktanakaj** [nestjs/swagger#140]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

We are trying to migrate our old express & swagger application to Nest.
The app have security parameter for own rule that is not bearer and oauth2.
But there are only @ApiBearerAuth() @ApiOAuth2Auth().

## Expected behavior

I guess we can use a generality security decorator. Example:
```
@ApiSecurity(securityDefinitionName)
@Get()
async findAll() {
```

## Environment

<pre><code>
Nest version: 5.3.7
@nestjs/swagger: 2.5.1

</code></pre>

**Comment by kamilmysliwiec** [comment]
Could you share your definition example?

**Comment by ktanakaj** [comment]
Sure. These are almost same definition.

https://github.com/ktanakaj/breakout-mk/blob/9e5df20b302cb4a9649ae5f2f261ec25e92e060b/breakout-svr/config/default.yaml#L304
https://github.com/ktanakaj/breakout-mk/blob/9e5df20b302cb4a9649ae5f2f261ec25e92e060b/breakout-svr/src/routes/users.ts#L71

I think it is very simple definition. But I couldn't find how to decorate by @nestjs/swagger...

**Comment by nartc** [comment]
@kamilmysliwiec looks like it is just a BearerAuth with a custom name. Once this is solved (https://github.com/nestjs/swagger/issues/130) then you can decorate your SessionID Auth with NestJS @ktanakaj

**Comment by ktanakaj** [comment]
Oh, great! I'll try to use BearerAuth with a custom name after #130 is solved. Thanks a lot!

**Comment by nartc** [comment]
@ktanakaj PR has been opened. Hopefully it gets reviewed and merged soon

**Comment by ktanakaj** [comment]
I was able to add my security definition by new `@ApiSecurity` tag on `@nestjs/swagger@next`. Thank you so much :+1:

**Comment by kamilmysliwiec** [comment]
4.0.0 has been published

**Comment by rmainseas** [comment]
Are there some release notes somewhere we can read? My project is on 3.1.x so I don't have access to `@ApiSecurity` - but it would be nice to have release notes so I can get a general flavor of how compatible 4.x is with 3.x.
